treeherder/log_parser/parsers.py

```diff
@@ -422,6 +426,15 @@ def add(self, line, lineno):
 
     def parse_line(self, line, lineno):
         """Check a single line for an error.  Keeps track of the linenumber"""
+        # First line of TaskCluster logs almost certainly has this.
+        if line.startswith('[taskcluster '):
+            self.is_taskcluster = True
+
+        # For performance reasons, only do this if we have identified as
+        # a TC task.
```

Comment:
On a Windows build log, the presence of the regular expression appears to add noticeable overhead. We went from ~0.92-0.98s to 1.04-1.06s. The added ".startswith()" doesn't appear to add noticeable overhead. This makes sense: the check will fail as early as examining the first character. Comparing a byte is much faster than invoking the regexp engine.